lib/vehicle.rb

```diff
@@ -5,14 +5,22 @@ class Vehicle
               :year,
               :make,
               :model,
-              :engine
+              :engine,
+              :registration_date
+
+ 
 
   def initialize(vehicle_details)
     @vin = vehicle_details[:vin]
     @year = vehicle_details[:year]
     @make = vehicle_details[:make]
     @model = vehicle_details[:model]
     @engine = vehicle_details[:engine]
+    @registration_date = vehicle_details[:registration_date]
+  end
+
+  def update_registration_date
+    @registration_date = Date.today
   end
```

Comment:
I like this method. I personally don't have an issue with `attr_accessors` however this gives us more control if we ever want to add in validation before setting the date. It's also a little more readable than using an accessor which is nice.